In [6]:
##text/string
print("hello world")

hello world


In [2]:
# Integers
print(8)

8


In [3]:
#decimal
print(8.44)

8.44


In [2]:
#boolean
print(True)
print(False)

True
False


In [8]:
#complex
print(4+3j)


(4+3j)


In [9]:
#tuple
print((1,2,3,4,))

(1, 2, 3, 4)


In [10]:
#list
print([1,2,3,4])

[1, 2, 3, 4]


In [11]:
#sets
print({1,2,3,4})

{1, 2, 3, 4}


In [2]:
#dictionary
print({'name':'nitish','gender':'male','weight':'70'})

{'name': 'nitish', 'gender': 'male', 'weight': '70'}


In [ ]:
#type (define the type of function just put the fn run it and it will show the type of fn )
type({1,2,3})

set

In [21]:
# Variables
name="amit" 
print(name,end='-')
a=("ajit")
b=("janu")
print(a + b)

amit-ajitjanu


In [ ]:
#dynamic typing (python)
a=4
#static typing(c++,java)
int a=4

In [ ]:
#dynamic binding(python support)
a=5 
print(a)
a='amit'
print(a)
b=5
print(b)

5
amit
5


In [ ]:
#static binding
int a=5


In [27]:
a=1
b=2
c=3
print(a,b,c)

1 2 3


In [28]:
a,b,c=1,2,3
print(a,b,c)

1 2 3


In [ ]:
a=b=c=5
print(a,b,c,)

5 5 5


comments

In [3]:
# this is a comment 
a=4
b=3
# second comment
print(a+b)


7


keywords

identifiers

In [ ]:
#identifiers
# You can't start with a digit
name="amit"
print(name)
#you can use special chars -> _
_='ajit'
print(_)

# identifiers can not be keywords

amit
ajit


user input

In [37]:
# static software vs dynmaic software
input()

'hello'

In [39]:
input('enter email')

''

In [8]:
  # Take input from user and store them in variables 
fnum=int(input('enter first number'))
snum=int(input("enter second number"))
   # add the 2 variables 
print(fnum,snum)
   
result= fnum + snum
   # print the result
print(result)

33 43
76


Type conversion 

In [ ]:
# Implicit and explicit
print(5+5.6)

print(type(5),type(5.6))

print(4+int('4'))

10.6
<class 'int'> <class 'float'>
8


This automatic conversion is called Implicit Type Conversion. (10.6)

Explicit 

Explicit = You tell Python what to do.

In [49]:
# Explicit
# str--> int
int('4')
type(int('4'))

int

Literals

What is a Literal?

A literal is a fixed value that you write directly in your code

In [ ]:
a=0b1010 #binary literal
b=100    # decimal literal (((( Decimal number system (base 10)  base 10 means Base 10 uses 10 digits:  0 1 2 3 4 5 6 7 8 9 That's why it's called base 10.))))
c=0o310  # Octal literal
d=0x12c  # Hexadecimal Literal

#float Literal   ((((The numbers with a decimal point (10.5, 3.14) are called float literals, not decimal literals)))).
float_1=10.5
float_2=1.5e2   # 1.5*10^2
float_3=1.5e-3  # 1.5*10^-3


#complex Literal
x= 3.14j

print(a,b,c,)
print(float_1,float_2)
print(x,x.imag,x.real)

In [ ]:
string='this is python'
strings="this is python"
char="C"
multiline_str="""This is a multiliner string more than one line code"""
unicode =u"\U001f600\U0001J43"
raw_str=r"raw \n string"

print(string)
print(string)
print(char)
print(multiline_str)
print(unicode)
print(craw_str)

SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 0-8: truncated \UXXXXXXXX escape (3909300518.py, line 5)

boolean 
(treat true as 1 
       false as 0 )

In [56]:
a=True+4
b=False+10

print("a",a)
print("b",b)

a 5
b 10


None

In [57]:
k=None
a=4
b=5
print("execute code")


execute code
